util/type_traits.h

```diff
@@ -50,12 +50,9 @@ using has_atomic_support = contains<T, int, unsigned int, long, unsigned long,
  * @brief Checks whether T is a floating-point sycl type
  */
 template <typename T>
-using is_cl_float_type =
-    std::bool_constant<std::is_floating_point<T>::value ||
-                       std::is_same<sycl::half, T>::value ||
-                       std::is_same<sycl::cl_float, T>::value ||
-                       std::is_same<sycl::cl_double, T>::value ||
-                       std::is_same<sycl::cl_half, T>::value>;
+using is_sycl_floating_point_type =
```

Comment:
I just noticed that there is `is_sycl_floating_point`, I think this alias can removed completely.
Please, update the code using `is_cl_float_type`.

Reply:
BTW, what is the reason of this change? Can we use `is_sycl_floating_point` instead of `is_cl_float_type`? I see that `is_cl_float_type` is still used by legacy tests and I suppose might be required to validate the compatibility with SYCL 1.2.1 version.

Reply:
> BTW, what is the reason of this change?

I faced an issue with hipsycl compilation in #318. The error was saying that there are no cl_* types,  and I didn't found these types in [paragrarph 4.14.1](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#_scalar_data_types) so I thought we can get rid of them

Reply:
Replaced old alias names

Reply:
Thanks. I hope we will not end-up with removing `cl_float` and `cl_double` types from the test. :)